nservicebus/outbox/index.md

```diff
@@ -79,8 +79,61 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. When considering not using the outbox and implementing idempotent processing keep in mind that any generated message *must* apply the same message identifier if invoked more-then-once to have deduplication working on the infrastructure level or to solve this at the logical level by validating message data. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
+
+
+## Concurrency
+
+Outbox cannot guarantee that handlers or sagas are invoked exactly once. Outbox does not prevent concurrent attempts to process the same message but ensures that the outcome of the processing is persisted only once for resources that share the outbox transaction.
```

Comment:
> Outbox cannot guarantee that handlers or sagas are invoked exactly once.

this still seems to be a contradiction to what this document states at the beginning:

> [...] At the handler level, however, the Outbox guarantees `exactly-once` message processing [...]

Reply:
if I understand this correctly, this is more of an issue where outbox cannot deduplicate two different physical but isomorphic messages ?

Reply:
@timbussmann To what do you refer?

Reply:
> this is more of an issue where outbox cannot deduplicate two different physical but isomorphic messages ?

No, also the same physical copy in case the handler is invoked but the outbox record not yet committed and the message to be reprocessed. Better said, outbox does not guarantee at-most-once execution, only transactionally exact-once or non-transactional at-least-once.

Reply:
This section is not a contradiction. The start of the document is about comparing transactional semantics of MSDTC with Outbox and for both they are transactionally exact-once. It even explicitly mentions that this is exact-once at 'a higher level'. On a low level it is still at-least-once processing hence this section explaining that it still is at-least-once.

Reply:
> Better said, outbox does not guarantee at-most-once execution, only transactionally exact-once or non-transactional at-least-once

If it's better said that way, then I'd suggest re-writing the section to say it that way.

I can see how this section could be construed as contradictory. If these are two different situations then you need to be clear, in both locations, which situation you're talking about. i.e. if the first section is talking about MSDTC with Outbox, then make it clear by using that phrase or those proper nouns.

Reply:
Changed the beginning to reflect this.